starter-code/src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+};
+
+ Soldier.prototype.attack = function() {
+    return this.strength;
+};
+
+Soldier.prototype.receiveDamage = function (damage){
+    
+    this.health = this.health - damage;
+};
+
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function (damage){
+    
+    this.health = this.health - damage;
+
+    if(this.health > 0) {
+        return this.name + ' has received ' + damage + ' points of damage';
+    } else {
+        return this.name + ' has died in act of combat';
+    }
+
+};
+
+Viking.prototype.battleCry = function (){
+    return "Odin Owns You All!";
+
+}
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+Saxon.prototype.receiveDamage = function (damage) {
+   
+    this.health = this.health - damage;
+
+    if(this.health > 0) {
+        return 'A Saxon has received ' + damage + ' points of damage';
+    } else {
+        return 'A Saxon has died in combat';
+    }
+}
 
 // War
-function War() {}
+function War() {
+
+   this.vikingArmy = [];
+   this.saxonArmy = [];
+
+}
+
+
+War.prototype.addViking = function (viking) {
+
+    this.vikingArmy.push(viking);
+
+
+}
+
+
+War.prototype.addSaxon = function (saxon) {
+
+    this.saxonArmy.push(saxon);
+
+}
+
+
+War.prototype.vikingAttack = function () {
+   var randomViking = this.vikingArmy[Math.floor(Math.random()*this.vikingArmy.length)];
+   var randomSaxon = this.saxonArmy[Math.floor(Math.random()*this.saxonArmy.length)];
+   var index = this.saxonArmy.indexOf(randomSaxon)
+    randomSaxon.receiveDamage(randomViking.strength)
+   
+    if(randomSaxon.health <= 0){
+     this.saxonArmy.splice(index, 1);
+     return 'A Saxon has died in combat';
```

Comment:
Instead of returning that string here (then you miss the return when the saxon survives), store the result of receiveDamage() in a variable and return that at the end (line 97).